jsonnet/docker/plugin.libsonnet

```diff
@@ -0,0 +1,117 @@
+local windows_pipe = '\\\\\\\\.\\\\pipe\\\\docker_engine';
+local windows_pipe_volume = 'docker_pipe';
+
+local windows_settings = {
+  daemon_off: true,
+  // Workaround for https://github.com/drone/drone-cli/issues/117
+  purge: 'false',
+};
+
+{
+  /**
+   * Returns a Docker plugin using the given settings.
+   *
+   * @param name The name of the pipeline step.
+   * @param settings The settings for the plugin.
+   * @return A json representing a Docker plugin and its settings.
+   */
+  Plugin(name, settings): {
+    name: name,
+    image: 'plugins/docker',
+
+    settings: settings,
+  },
+
+  /**
+   * Returns a Docker plugin for Windows builds.
+   *
+   * Currently Windows does not support docker in docker so the pipe must be
+   * mounted as a volume. This requires trusted builds to be enabled for the
+   * repository.
+   *
+   * The pipeline must have
+   *
+   * @param name The name of the pipeline step.
+   * @param settings The settings for the plugin.
+   * @return A json representing a Docker plugin and its settings.
+   */
+  WindowsPlugin(name, settings): self.Plugin(name, settings + windows_settings) {
```

Comment:
But what you're doing here (applying windows_settings on top) can only be done with a function param. Using mixins you can only have windows_settings underneath.